src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -3,40 +3,113 @@
+        this.index = -1; // not used
+    }
+
+    /**
+     * Constructs a FindCommand object which searches for the person at a specific index.
+     *
+     * @param index The index that the user searched for.
+     */
+    public FindCommand(int index) {
+        this.namePredicate = null;
+        this.index = index;
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+
+        if (namePredicate != null) {
+            return executeNameSearch(model);
+
+        } else if (index > -1) {
+            checkIndex(model); // throws an exception if index is out of range
+            indexPredicate = new StaffHasCorrectIndexPredicate(index, model);
+            return executeIndexSearch(model);
+
+        } else {
+            throw new CommandException("Check if your input are correct: -n for name, -i for index,\n"
+                    + "and that the index given is correct!");
+        }
+    }
+
+    /**
+     * Executes a search by name.
+     *
+     * @param model The model which contains the list to be searched on.
+     * @return a CommandResult to be displayed.
+     */
+    public CommandResult executeNameSearch(Model model) {
+        model.updateFilteredPersonList(namePredicate);
         return new CommandResult(
                 String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
     }
 
+    /**
+     * Executes a search by index.
+     *
+     * @param model The model which contains the list to be searched on.
+     * @return a CommandResult to be displayed.
+     */
+    public CommandResult executeIndexSearch(Model model) {
+        model.updateFilteredPersonList(indexPredicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+    }
+
+    /**
+     * Checks if the index is within a suitable range for the list contained in the model.
+     *
+     * @param model The model which contains the list to be searched on.
+     * @throws CommandException When the index inputted is not within range.
+     */
+    public void checkIndex(Model model) throws CommandException {
+        int personListSize = model.getAddressBook().getPersonList().size();
+        if (index > personListSize) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+    }
+
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof FindCommand // instanceof handles nulls
-                && predicate.equals(((FindCommand) other).predicate)); // state check
+                && (namePredicate == null || namePredicate.equals(((FindCommand) other).namePredicate))
```

Comment:
Wouldn't this mean that a find command with a name predicate and a find command with an index predicate are equal?
```suggestion
                && (namePredicate.equals(((FindCommand) other).namePredicate)
                && (namePredicate.index == ((FindCommand) other).index)
                
```